diff --git a/node_modules/dockerfilelint/lib/checks.js b/node_modules/dockerfilelint/lib/checks.js
index b49bbf0..d70512d 100644
--- a/node_modules/dockerfilelint/lib/checks.js
+++ b/node_modules/dockerfilelint/lib/checks.js
@@ -181,12 +181,12 @@ var commands = module.exports = {
       "--retries"];
     var cmd = /\ [^-].*/.exec(args);
     if (cmd != null) {
-      var options = args.substring(0, cmd.index).match(/--\w+/g);
+      var options = args.substring(0, cmd.index).match(/--[\w-]+/g);
       if (!options.reduce((valid, item) => valid &&
         (allowedoptions.indexOf(item) > -1), true)) return ['invalid_format'];
 
       if (options != null) {
-        var optparams = args.substring(0, cmd.index).match(/--\w+=\d+\w*/g);
+        var optparams = args.substring(0, cmd.index).match(/--[\w-]+=\d+\w*/g);
         return ((optparams != null) &&  (optparams.length == options.length)) ?
           [] : ['healthcheck_options_missing_args'];
       }
